### ISSUE TITLE
[CS] characters comparison should be unicode-sensitive (in contrast to culture/word-sensitive)

### ISSUE THREAD

**Issue by azrafe7** [HaxeFoundation/haxe#6717]
Code:

```haxe
  var hyphen = "-";
  var space = " ";

  trace(hyphen.charAt(0) > space.charAt(0)); // false on CS, true on other targets (including java)
```

Only CS seems to be affected (and probably it extends to strings too).

**Comment by azrafe7** [comment]
Well, to _some_ C# strings...

```haxe
  trace("coop" > "co-op"); // false on CS, true otherwise
```

I've found some insigths here https://msdn.microsoft.com/en-gb/library/system.globalization.compareoptions(v=vs.110).aspx#Anchor_2

**Comment by bendmorris** [comment]
Does Haxe specify String comparisons, or just defer to the target behavior? If it's defined behavior these would be good test cases.

**Comment by nadako** [comment]
Closed by #7562, right?
